cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +795,88 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+const char* gdv_fn_mask_first_n(int64_t context, const char* data, int32_t data_len,
+                                int32_t n_to_mask, int32_t* out_len) {
+  if (data_len <= 0) {
+    *out_len = 0;
+    return nullptr;
+  }
+
+  if (n_to_mask < 0) {
```

Comment:
Not sure about this. Can you check how this is implemented in Hive?

Reply:
In Hive implementation it did not say anything about that, but I've changed the code and now it handles this case better